File: Assets/Scripts/Lua/UnityLuaAPI/MFmodVCA.lua
---@class MFmodVCA
---@field public vca FMOD.Studio.VCA
---@field public Deprecated boolean

---@type MFmodVCA
MFmodVCA = { }
---@return MFmodVCA
---@param vca FMOD.Studio.VCA
function MFmodVCA.New(vca) end
---@return System.Guid
function MFmodVCA:getID() end
---@return string
function MFmodVCA:getPath() end
---@param volume System.Single
---@param finalvolume System.Single
function MFmodVCA:getVolume(volume, finalvolume) end
---@param volume number
function MFmodVCA:setVolume(volume) end
return MFmodVCA

File: Assets/Scripts/FMod/MFmodVCA.cs
using System;
using FMOD.Studio;
using MoonCommonLib;

public class MFmodVCA : IMFmodVCA
{
    public VCA vca { get; private set; }

    public bool Deprecated { get; set; }

    public MFmodVCA(VCA vca)
    {
        this.vca = vca;
    }

    public Guid getID()
    {
        Guid id;
        vca.getID(out id);
        return id;
    }

    public string getPath()
    {
        string path;
        vca.getPath(out path);
        return path;
    }

    public void getVolume(out float volume, out float finalvolume)
    {
        vca.getVolume(out volume, out finalvolume);
    }

    public void setVolume(float volume)
    {
        vca.setVolume(volume);
    }
}